mesonbuild/build.py

```diff
@@ -34,7 +34,7 @@
 )
 from .compilers import (
     Compiler, is_object, clink_langs, sort_clink, lang_suffixes,
-    is_known_suffix
+    is_known_suffix, detect_static_linker
```

Comment:
I think this probably belongs in linkers, but since I wanted to split the dynamic and static linkers anyway I'll do that after this lands.

Reply:
Yeah, but `linkers.py` is already pretty big so the detection logic shouldn't go there. So it would probably be a good idea to make a new `mesonbuild.linkers` module instead. But then we have to maintain another `__init__.py` for just two files, so I don't really know.

Reply:
OK, I have moved *most* of the linker stuff from `compilers.detect` to `linkers.detect`. However, the `detect_static_linker` function is too tightly coupled with the compilers logic to sensibly move it somewhere else. I guess I will leave refactoring this to you then :)

Reply:
That's fine, I seem to have a thing for doing awful refactors :)